glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -159,19 +186,20 @@ public class HttpPingUploader {
                     Dispatchers.shared.launchAsync {
                         HttpPingUploader(configuration: self.config, testingMode: self.testingMode).process()
                     }
+                    uploadGroup.leave()
```

Comment:
I always forget `defer` even exists...   I mostly see it used more like a `try/finally` but yeah, it would work here.

Reply:
This only solves half the problem though.
In this `case` we're within the closure used by `self.upload`.
In the `.wait` and `.done` further below (lines 194ff) we're outside of it. 

So regardless we wil need one defer in there, but still the individual calls further below (because we don't want to double-fire it)